### PR TITLE
add crane addon

### PR DESCRIPTION
**Pull request by fengkang01**
### Description of your changes
add crane addon

### Checklist

I have:

- [ ] Title of the PR starts with type (e.g. `[Addon]` or `[Trait]`).
- [ ] Updated/Added any relevant [documentation] and [examples].
- [ ] Unit/E2E Tests added.

### CODE REVIEW

**Review by wangyikewxgm — COMMENTED** [review]
Please move this addon to experimental addons.

**Comment by wangyikewxgm** [comment]
Please pending this pr until support read value from configmap. Then refactor this addon with helm type component.

**Review by oeular — COMMENTED** [review]
Pls changes the yaml file in resources with cue, and expose the related paramters for user to controller.

**Review by oeular — COMMENTED** [review]
Pls give some examples in README.md for more users to use it normally. (you'd like change readme.md to README.md)

**Comment by fengkang01** [comment]
about the crane addon ,the crane official provide installation not have other extension at now.About the thing that change yaml file to cue file,i want change when the vela helm addon support the valueFrom parameter.at that time i will change the crane addon use cue file and i will change this addon to use helm addon rather than k8s api about (HelmRepository,HelmRelease)

**Comment by fourierr** [comment]
It is amazing and exciting to integrate the crane as an vela addon for me. Later I will try it out.

**Comment by fourierr** [comment]
> about the crane addon ,the crane official provide installation not have other extension at now.About the thing that change yaml file to cue file,i want change when the vela helm addon support the valueFrom parameter.at that time i will change the crane addon use cue file and i will change this addon to use helm addon rather than k8s api about (HelmRepository,HelmRelease)

About changing yaml file to cue file previously mentioned , just follow the two principles
- The YAML type file should contain a K8S resource object, which will be directly added to the application as a K8s-object type component during rendering
- If you need to add a component to your application that needs to be rendered dynamically based on parameters when enabled, you can write a CUE format file. ping @fengkang01

**Comment by wangyikewxgm** [comment]
https://docs.gocrane.io/dev/tutorials/using-effective-hpa-to-scaling-with-effectiveness/

Can we make this capability as a trait?

File: experimental/addons/crane/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: crane
+version: v0.5.0
+description: crane
+icon: https://docs.gocrane.io/dev/images/crane.svg
+url:
+
+tags:
+  - crane
```
Comment (wangyikewxgm) [inline experimental/addons/crane/metadata.yaml:8]: - FinOps

**Comment by wangyikewxgm** [comment]
i think this addon is a good practice in finOps area. Let's find out if  [recommendation](https://docs.gocrane.io/dev/tutorials/hpa-recommendation/) and [HPA](https://docs.gocrane.io/dev/tutorials/using-effective-hpa-to-scaling-with-effectiveness/) can use as a vela trait.

**Review by wangyikewxgm — APPROVED** [review]
LGTM，Good job. Will add more trait capability in next pr.

**Comment by wonderflow** [comment]
let's refine the addon in the following PRs.

**Pull request merged**